### DISCUSSION TITLE
Detect language using Google Translate

### DISCUSSION THREAD

**Discussion by chbridges** [nidhaloff/deep-translator#153] [Q&A]
There are some (no longer maintained) modules for free Google Translate usage which also use Google Translate for language detection.

Would this be a possible addition to Deep-Translator in order to bypass the limit of the free DetectLanguage API key?
Would it even be efficient or would it come with a similar overhead as the translation?

For context, I'm translating texts to English with Google Translate (which works great with Deep-Translator!) but want to filter out English texts before doing so.

**Comment by nidhaloff** [comment]
@chbridges Unfortunately this is not possible because deep-translator is using web scraping and not the official google API.
